spring-boot-security/src/test/java/com/baeldung/springbootsecurity/oauth2server/OAuth2IntegrationTestSupport.java

```diff
@@ -31,4 +39,18 @@ protected OAuth2RestTemplate getOAuth2RestTemplate(final ClientCredentialsResour
         restTemplate.setMessageConverters(singletonList(new MappingJackson2HttpMessageConverter()));
         return restTemplate;
     }
+
+    protected String executeGetRequest(OAuth2RestTemplate restTemplate, String path) {
+        return restTemplate.execute(base.toString() + path, GET, getRequestCallback(), getStringResponseExtractor());
+    }
+
+    protected RequestCallback getRequestCallback() {
+        return clientHttpRequest -> {
+        };
+    }
+
+    protected ResponseExtractor<String> getStringResponseExtractor() {
+        return clientHttpResponse -> IOUtils.toString(clientHttpResponse.getBody(), Charset.defaultCharset());
+    }
```

Comment:
Similarly, you don't need to make a new lambda every time for this. You just want something like:

```java
protected static final ResponseExtractor<String> EXTRACT_BODY_AS_STRING =  clientHttpResponse -> IOUtils.toString(clientHttpResponse.getBody(), Charset.defaultCharset());
```